qt/python/mantidqt/widgets/fitpropertybrowser/fitpropertybrowserplotinteraction.py

```diff
@@ -0,0 +1,308 @@
+        """
+        Toggle the guess all line
+        """
+        if self.guess_all_line:
+            self.remove_guess_all()
+        else:
+            self.plot_guess_all()
+
+    @Slot()
+    def plot_current_guess_slot(self):
+        """
+        Plots the guess workspace for the current function,
+        which is currently selected Handler
+        """
+        self.fit_browser.currentHandler().setHasPlot(True)
+        self.plot_current_guess()
+
+    @Slot()
+    def remove_current_guess_slot(self):
+        """
+        Removes the guess workspace for the current function,
+        which is currently selected Handler
+        """
+        self.fit_browser.currentHandler().setHasPlot(False)
+        self.remove_current_guess()
+
+    @Slot(str)
+    def parameters_changed_slot(self, prefix):
+        """
+        Responds to parameters being changed in the FitPropertyBrowser
+        :param prefix of the function which the parameters have been changed (unused)
+        """
+        if not self.fit_browser.currentHandler():
+            return
+        name = self._get_current_prefixed_function_name()
+        if name in self.guess_lines:
+            self.update_current_handler_guess()
+        if self.guess_all_line:
+            self.update_guess_all()
+
+    @Slot()
+    def slot_for_function_removed(self):
+        """
+        Responds to a function being removed from the FitPropertyBrowser,
+        the function is found from the currently selected handler
+        If the function, or guess all is plotted the plot is updated accordingly
+        """
+        if not self.fit_browser.currentHandler():
+            return
+        prefix = self.fit_browser.currentHandler().functionPrefix()
+        self.remove_current_guess()
+        self._shift_stored_prefixes(prefix)
+        if self.guess_all_line:
+            self.update_guess_all()
+
+    def plot_current_guess(self):
+        """
+        Plot the guess workspace of the currently selected function
+        """
+        fun = self.fit_browser.currentHandler().ifun()
+        ws_name = self.fit_browser.workspaceName()
+        if fun == '' or ws_name == '':
+            return
+        out_ws_name = self._get_current_prefixed_function_name()
+
+        line = self._plot_guess_workspace(ws_name, fun, out_ws_name)
+
+        if line:
+            self.guess_lines[self._get_current_prefixed_function_name()] = line
+
+    def evaluate_function(self, ws_name, fun, out_ws_name):
+        """
+        Evaluates the guess workspace for the input workspace and function
+        :param ws_name: Name of the workspace in the ADS
+        :param fun: Function to be evaluated
+        :param out_ws_name: Output workspace name
+        :return: Output guess workspace
+        """
+        ws_index = self.fit_browser.workspaceIndex()
+        startX = self.fit_browser.startX()
+        endX = self.fit_browser.endX()
+        workspace = AnalysisDataService.retrieve(ws_name)
```

Comment:
Would a check to see if the workspace exists in the ADS before this line be important here?

Reply:
As this is the same workspace thats in the plot we can be pretty certain it exists. If its not in the ads (e.g if we delete it while having a plot shown) it appears to close the plot.